### PR TITLE
Don't load large amount of objects into memory. Load their ids instead.

### PR DESCRIPTION
**Pull request by phanle**
This is a simple version but can run to the risk of max packet size for
a long query.
We can improve after this.

### CODE REVIEW

File: lib/broken_record/job.rb [hunk 1]
```diff
@@ -57,10 +57,9 @@ def records
         klass.unscoped
       end
 
-      records_per_group = model_scope.count / self.class.jobs_per_class
+      records_per_group = (model_scope.count / self.class.jobs_per_class.to_f).ceil
       scope = model_scope.offset(records_per_group * index)
-
-      index == Parallel.processor_count - 1 ? scope.load : scope.first(records_per_group)
+      model_scope.where("#{klass.table_name}.#{klass.primary_key}" => scope.limit(records_per_group).pluck(:id))
```
Comment (nickgervasi) [inline lib/broken_record/job.rb]: Do we need to `pluck` the `primary_key` instead of `:id`?

File: lib/broken_record/job.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def perform
       result = BrokenRecord::JobResult.new(self)
       result.start_timer
 
-      records.each do |r|
+      records.find_each do |r|
```
Comment (nickgervasi) [inline lib/broken_record/job.rb]: One way to mitigate the max packet size concern is to slice the ids into batches of say 1000 records using `each_slice`. I think it's worth doing now vs. waiting until later.

**Comment by phanle** [comment]
Hey @nickgervasi I changed the code as per you comment.
Have a look and let me know

**Comment by phanle** [comment]
Bumped this.

**Pull request merged**